alarm_active_table.cpp

```diff
@@ -0,0 +1,423 @@
+    }
+    break;
+    
+    case COLUMN_ALARMACTIVEDESCRIPTION:
+    {
+      snmp_set_var_typed_value(var, ASN_OCTET_STR,
+                               (u_char*) context->_alarm_table_def->description().c_str(),
+                               context->_alarm_table_def->description().length());
+    }
+    break;
+
+    case COLUMN_ALARMACTIVELOGPOINTER:
+    {
+      // We don't keep log entries so this will always be
+      // the 0.0 OID
+      snmp_set_var_typed_value(var, ASN_OBJECT_ID,
+                               (u_char*) zero_dot_zero_oid,
+                               sizeof(zero_dot_zero_oid));
+    }
+    break;
+
+    case COLUMN_ALARMACTIVEMODELPOINTER:
+    {
+      oid* model_pointer;
+      // Convert the index string OID
+      int model_pointer_len = alarmModelTable_oid_len + entry_column_oid_len + 3;
+      // Append the index array on to the Alarm Model Table OID, inserting
+      // the OID ".1.3." for the entry in the first non index column
+      model_pointer = new oid[model_pointer_len];
+      copy(alarmModelTable_oid, alarmModelTable_oid + alarmModelTable_oid_len, model_pointer);
+
+      copy(entry_column_oid, entry_column_oid + entry_column_oid_len, model_pointer + alarmModelTable_oid_len);
+      
+      model_pointer[alarmModelTable_oid_len + entry_column_oid_len] = 0;
+      model_pointer[alarmModelTable_oid_len + entry_column_oid_len + 1] = context->_alarm_table_def->alarm_index(); 
+      model_pointer[alarmModelTable_oid_len + entry_column_oid_len + 2] = context->_alarm_table_def->severity();      
+      
+      snmp_set_var_typed_value(var, ASN_OBJECT_ID,
+                               (u_char*) model_pointer,
+                               model_pointer_len * sizeof(oid));
+
+    }
+    break;
+    
+    case COLUMN_ALARMACTIVESPECIFICPOINTER:
+    {
+      // This would point to the model-specific active alarm list
+      // for this alarm. We do not support model-specific Alarm-MIB
+      // and hence this will always be the 0.0 OID
+      snmp_set_var_typed_value(var, ASN_OBJECT_ID,
+                               (u_char*) zero_dot_zero_oid,
+                               sizeof(zero_dot_zero_oid));
+    }
+    break;
+    
+    
+    default: /** We shouldn't get here */
+    {
+      TRC_ERROR("unknown column: alarmActiveTable_get_value");
+      return SNMP_ERR_GENERR;
+    }
+  }
+
+  return SNMP_ERR_NOERROR;
+}
+
+/************************************************************
+ * 
+ *  Create a new row context and initialize its oid index.
+ */
+void alarmActiveTable_create_row(char* name,
+                                 alarmActiveTable_SNMPDateTime* datetime,
+                                 unsigned long index,
+                                 AlarmTableDef& def)
+{
+  alarmActiveTable_context* ctx = SNMP_MALLOC_TYPEDEF(alarmActiveTable_context);
+  ctx->_alarm_table_def = &def;
+  if (pointers[ctx->_alarm_table_def->alarm_index()])
```

Comment:
It's not very clear what this code is doing. Comments would definitely help, as would clearer variable names:

```
alarmActiveTable_context* existing_row = alarm_indexes_to_rows[def.alarm_index()];
if (existing_row != NULL)
{
  if (def.severity == existing_row->_alarm_table_def->severity())
  {
    // Existing alarm has been re-raised, do nothing
    return;
  }
}
```

seems a bit clearer to me.